### ISSUE TITLE
Abstract render actions logic a little more

### ISSUE THREAD

**Issue by ScopeyNZ** [dnadesign/silverstripe-elemental#403]
Currently we choose to hide the actions menu of an element if the block is not "expandable":

https://github.com/dnadesign/silverstripe-elemental/blob/ff0e0b77bd340e4698b8338a54fcdb2f5de2f770/client/src/components/ElementEditor/Header.js#L99

Preferably we should change this to a separate flag and make it configurable (defaulting to whether the block is expandable). I expect some would prefer actions like `Publish` to be available on all blocks - although `Save` wouldn't really make sense on blocks that are not in-line editable.

Thoughts?

**Comment by robbieaverill** [comment]
Yep. Publish, unpublish and delete should be available even without inline editing enabled for the block (I think), unless @clarkepaul disagrees

**Comment by clarkepaul** [comment]
@ScopeyNZ Remind me about where we got to for dragging blocks up and down, does that effect the state of the dragged block. I think you were going to try to avoid the rest of the blocks states being effected but just the one which is being dragged would change to Draft/Modified (and page going into Draft/Modified also)?

If thats the case, I think the same actions are a available no matter if the block is expanded, collapsed, and even when non-inline editable, with the exception of not needing the Save action for the non-inline editable blocks. 

Very confusing terminology we are using, we need some better names for these things :)

**Comment by ScopeyNZ** [comment]
Regarding this issue, the designs currently have no actions (not even the meatballs) showing on non-in-line editable blocks. Should we change the default so they do show? Additionally, you mention not including "save" if it's not in-line editable but what about the tab buttons (Content, Settings etc...)?

Dragging is an issue that's not limited to Elemental. In our case we're just going to create a new version for the block that is being moved but feel free to weigh in on the conversation in this issue: https://github.com/silverstripe/silverstripe-versioned/issues/180

**Comment by clarkepaul** [comment]
Ah right, sorry I didn't refer to the designs, and I think I'm now up-to-date now. I would agree that adding those actions (Publish, Archive, Unpublish) would make sense. If its scope creep for what your trying to achieve then they could be excluded as per designs (pushed out into a seperate issue). If its a fit then we can add the dropdown to the designs specs. For non-inline editable blocks no "Save" action required and no tabs at this stage.

**Comment by brynwhyman** [comment]
Can I suggest we update this issue to include the configurable flag _and_ include Publish, Archive, and Unpublish by default for all non-in-line editable blocks. It seems to make sense to do this all at once instead of a halfway solution.

**Comment by ScopeyNZ** [comment]
Agreed. Updated 🙂

**Comment by clarkepaul** [comment]
Cool adding a design task to add it to the designs too.
